File: 2023/UICPC Regional/Round #15/solutions/gameofgnomes/jo.py
n,m,k=map(int,raw_input().split())
a=0
for r in range(n%k,min(n+1,m*k),k):
  v=r%m;a=max(a,((n-r)/k+1)*(n+r)-2*r+(r/m)*m*(m+1)+v*v+v)
print a/2

File: 2023/UICPC Regional/Round #15/solutions/hothike/jo.kt
import java.util.*
import kotlin.math.max

fun main() {
    val scan = Scanner(System.`in`)
    val n = scan.nextInt()
    val T = (0 until n).map{scan.nextInt()}
    var i0 = 0
    var min = Integer.MAX_VALUE
    for (i in 0..(n-3)) {
        val m = max(T[i], T[i+2])
        if (m < min) {
            i0 = i
            min = m
        }
    }
    println("${i0+1} $min")
}
